### ISSUE TITLE
Enable variable interpolation in the timezone field.

### ISSUE THREAD

**Issue by hartfordfive** [logstash-plugins/logstash-filter-date#15]
Would it be possible to add the ability to allow for variable interpolation in the **timezone** field of the date plugin?  This would be useful in multi-tenant setups where servers from multiple timezones might ship their logs to one central Logstash server/cluster which is located in a different timezone.  It also avoids having a large conditional statement that would include every potential timezone.

Example:

```
date {
        match => [
           "event_time",
           "yyyy/mm/dd HH:mm:ss", "dd/MMM/yyyy:HH:mm:ss Z", "EEE MMM dd HH:mm:ss yyyy" 
        ]
        locale => "en"
        add_tag => [ "timestamp_updated", "tz_%{timezone}" ]
        timezone => "%{timezone}"
      }
```

Right now, the following example simply causes this error:  ```The datetime zone id '%{timezone}' is not recognised```.

**Comment by markwalkom** [comment]
+1 on this, it'd be nice to be able to pull this from the event if it contains it in a separate field.

**Comment by purbon** [comment]
Fixed in #31
